embedded-graphics/src/geometry/point.rs

```diff
@@ -176,6 +184,24 @@ impl Point {
     pub(crate) fn component_max(self, other: Self) -> Self {
         Self::new(self.x.max(other.x), self.y.max(other.y))
     }
+
+    /// Returns the squared length.
+    ///
+    /// The returned value is the square of the length of a vector from `(0, 0)`
+    /// to `(self.x, self.y)`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust,ignore
+    /// use embedded_graphics::geometry::Point;
+    ///
+    /// let p = Point::new(3, 4);
+    ///
+    /// assert_eq!(p.length_squared(), 25);
+    /// ```
+    pub(crate) fn length_squared(self) -> i32 {
```

Comment:
It would indeed reduce the number of cast in our code, but I'm not sure if this is true for all code using this method. I would suggest to leave it as `i32`and add a note in #338 to decide this when we make the method public.